### PR TITLE
New PlugIn: SMA

### PR DESCRIPTION
**Pull request by Boernsman**
nymea-plugins pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

- [x] If you added a new plugin, should it be added to nymea-plugins-all or nymea-plugins-maker?

### CODE REVIEW

**Comment by t-mon** [comment]
I will re-tag experimental after the next builds. It's conflicting in the nymea-plugins.pro

**Comment by Boernsman** [comment]
@t-mon rebased to 26, please review.

File: sma/sunnywebbox.cpp [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            return;
+        }
+
+        if (map.contains("proc") && map.contains("result")) {
+            QString requestType = map["proc"].toString();
+            QString requestId = map["id"].toString();
+            QVariantMap result = map.value("result").toMap();
+            parseMessage(requestId, requestType, result);
+        } else if (map.contains("proc") && map.contains("error")) {
+        } else {
```
Comment (mzanetti) [inline sma/sunnywebbox.cpp:316]: There seems a bit of error handling missing.

**Review by mzanetti — COMMENTED** [review]
Looks mostly fine. The discovery method seems a bit troublesome:
a) if possible we should really try to identify things by an ID, not an IP
b) depending on nmap is not the most awesome thing but I'd accept it if nothing else can be used

**Pull request merged**